.github/workflows/integration_test.yaml

```diff
@@ -19,7 +19,26 @@ jobs:
             value: "--num-units=3"
           - name: num_units=1, db_from_relation
             value: "--num-units=1"
+        exclude:
```

Comment:
Out of curiosity, what does this `exclude` do?

Reply:
Exclude some combinations created by matrix. Some of the test arguments are redundant for `test_error` and `test_upgrade`, exclude those cases to save some resources.

Reply:
Makes sense

Reply:
Does the syntax for exclude need both name & value of the args to exclude to be repeated? Not familiar with exclude 😞

Reply:
Since the `args` is an object (dict) in matrix, it has to be an object in `exclude` too